### PR TITLE
Sensor fusion/new driving state update

### PR DESCRIPTION
**Pull request by jkerpe**
- used DrivingState (including Parametersets) of LineFollower Application in SensorFusion App
- Minor documentation updates

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -156,7 +156,7 @@ Example for the **LineFollowerTarget** application:
 | ConvoyFollower | Convoy follower, providing information to the [DroidControlShip](https://github.com/BlueAndi/DroidControlShip) to drive to its target. | No | Yes | ./webots/worlds/zumo_with_com_system/PlatoonTrack.wbt |
 | LineFollower | Just a line follower, using a PID controller. | Yes | No | ./webots/worlds/ETrack.wbt  ./webots/worlds/LargeTrack.wbt  ./webots/worlds/LineFollowerTrack.wbt |
 | RemoteControl | The robot is remote controlled by e.g. the [DroidControlShip](https://github.com/BlueAndi/DroidControlShip) in a convoy follower role. | No | Yes | ./webots/world/zumo_with_com_system/* |
-| SensorFusion | The robot provides odometry and inertial data to the [DroidControlShip](https://github.com/BlueAndi/DroidControlShip), which calculates the sensor fusion based location information. | No | Yes | ./webots/worlds/zumo_with_com_system/SensorFusionTrack.wbt |
+| SensorFusion | The robot provides odometry and inertial data to the [DroidControlShip](https://github.com/BlueAndi/DroidControlShip), which calculates the sensor fusion based location information. | No | Yes | ./webots/world/zumo_with_com_system/LineFollowerTrack.wbt |
```
Comment (gabryelreyes) [inline README.md:159]: Wrong path

File: lib/APPSensorFusion/src/DrivingState.cpp [hunk 2]
```diff
@@ -57,28 +58,57 @@
...
+static void logCsvData(const uint16_t* lineSensorValues, uint8_t length, int16_t pos, int16_t pos3, bool isPos3Valid);
+void        logCsvDataTrackStatus(DrivingState::TrackStatus trackStatus);
+static void logCsvDataSpeed(int16_t leftSpeed, int16_t rightSpeed);
+#endif /* DEBUG_ALGORITHM */
+
 /******************************************************************************
  * Local Variables
  *****************************************************************************/
 
+#ifdef DEBUG_ALGORITHM
```
Comment (gabryelreyes) [inline lib/APPSensorFusion/src/DrivingState.cpp:73]: Is it intended to include all there DEBUG blocks?

File: lib/APPSensorFusion/src/ErrorState.h [hunk 3]
```diff
@@ -45,6 +45,7 @@
  *****************************************************************************/
 #include <stddef.h>
 #include <IState.h>
+#include "StartupState.h"
```
Comment (gabryelreyes) [inline lib/APPSensorFusion/src/ErrorState.h:48]: Maybe move this to cpp?

File: webots/protos/SensorFusionTrack.proto [hunk 4]
```diff
@@ -13,7 +13,7 @@ PROTO SensorFusionTrack [
     floorAppearance PBRAppearance {
       baseColorMap ImageTexture {
         url [
-          "./SensorFusionTrack.png"
+          "./track.png"
```
Comment (gabryelreyes) [inline webots/protos/SensorFusionTrack.proto:16]: Is this correct? Are you using the normal Track?

**Review by gabryelreyes — CHANGES_REQUESTED** [review]

File: lib/APPSensorFusion/src/ErrorState.h (hunk 3 above)
Comment (jkerpe) [reply lib/APPSensorFusion/src/ErrorState.h:48]: I got an error when trying to build it when the include is inside the cpp:
`error: 'M_PI' was not declared in this scope`

File: lib/APPSensorFusion/src/DrivingState.cpp (hunk 2 above)
Comment (jkerpe) [reply lib/APPSensorFusion/src/DrivingState.cpp:73]: removed

File: webots/protos/SensorFusionTrack.proto (hunk 4 above)
Comment (jkerpe) [reply webots/protos/SensorFusionTrack.proto:16]: yes, I use the regular linefollower track.

**Pull request merged**